### PR TITLE
Merge 2.6.6 changes into 2.7.4

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: ADAL/src/validation/ADDrsDiscoveryRequest.m [hunk 1]
```diff
@@ -30,13 +30,15 @@ @implementation ADDrsDiscoveryRequest
 + (void)requestDrsDiscoveryForDomain:(NSString *)domain
                             adfsType:(AdfsType)type
                              context:(id<MSIDRequestContext>)context
+                     requestMetadata:(NSDictionary *)metadata
```
Comment (jasoncoolmax) [inline ADAL/src/validation/ADDrsDiscoveryRequest.m]: Just curious, do we still need this given that `MSIDRequestContext` already contains `appRequestMetadata`? :)

Comment (oldalton) [reply ADAL/src/validation/ADDrsDiscoveryRequest.m]: I don't think so, will update :)

Comment (oldalton) [reply ADAL/src/validation/ADDrsDiscoveryRequest.m]: Addressed.

File: ADAL/src/validation/ADAuthorityValidation.m [hunk 2]
```diff
@@ -381,12 +381,12 @@ - (void)validateADFSAuthority:(NSURL *)authority
 }
 
 - (void)requestDrsDiscovery:(NSString *)domain
-                    context:(id<MSIDRequestContext>)context
+              requestParams:(ADRequestParameters *)requestParams
```
Comment (jasoncoolmax) [inline ADAL/src/validation/ADAuthorityValidation.m]: Just curious, any reason it needs to be `ADRequestParameters` but not `id<MSIDRequestContext>`? :)

Comment (oldalton) [reply ADAL/src/validation/ADAuthorityValidation.m]: Nope, it got merged from 2.6.6 :) Fixed it.

**Pull request merged**